### PR TITLE
fix: resolve layout issues with sticky footer forms inside Carbon modal components (FE-6816)

### LINKED ISSUE #6969: [Dialog] - Long title makes dialog content to overflow in small screens
### Description

Given a Dialog component  that has a Form as a children with the stickyFooter prop
When rendered in a mobile screen width
Then the dialog content is push down and overflows the dialog boundaries.

### Reproduction

https://stackblitz.com/edit/parsium-carbon-starter-h5jjs3?file=src%2FApp.tsx

### Steps to reproduce

1. Open the link https://stackblitz.com/edit/parsium-carbon-starter-h5jjs3?file=src%2FApp.tsx
2. Scroll down the dialog content

### JIRA ticket numbers (Sage only)

_No response_

### Suggested solution

Given a Dialog component  that has a Form as a children with the stickyFooter prop
When rendered in a mobile screen width
Then the dialog content is stays inside the dialog

### Carbon version

142.9.1

### Design tokens version

4.29.0

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Additional context

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by Parsium**
closes #6969

### Proposed behaviour

  - Address overflow issue when using `Form` with a sticky footer in:
       - `Dialog`
       - `DialogFullscreen`
       - `Sidebar`
       Screenshot 2024-10-09 at 16 57 55
  - Utilise CSS Flexbox more internally - to make layouts more flexible to viewport size changes 
  - Add stories for checking the issue hasn't regressed
  - Ensure `Dialog` height never exceeds 90% of the viewport height

### Current behaviour

- When a sticky footer `Form` is placed inside a `Dialog`, `DialogFullscreen` and `Sidebar`, it's possible for the form's content to overflow the modal at small screen sizes:
371104967-d8affa4e-59ef-4cc4-a1a8-db1b8014a79a

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- Please check the following stories when checking the overflow bug has been fixed:
  - `Dialog`
    - Test/Dialog With Long Header Content
  - `DialogFullscreen`
    - Test/With Sticky Form
  - `Sidebar`
    - Test/With Sticky Form
    - Test/With Form

### CODE REVIEW

**Review by DipperTheDan — APPROVED** [review]

File: src/components/form/form.stories.tsx
Comment (DipperTheDan) [inline src/components/form/form.stories.tsx:1]: **issue:** I've noticed a bug with Form. The left and right borders have been cut off when the textbox is focused. The screenshot is from the default test story.

Screenshot 2024-10-04 at 09 35 52

**Review by edleeks87 — COMMENTED** [review]
Good work @Parsium I really like the CSS solution over the previous observer approach but we may need to tweak it slightly so the transform isn't blurring the outlines when it gets a floating pixel value

**Review by edleeks87 — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Comment by harpalsingh** [comment]
@Parsium looks good to me.

**Review by paulrobinson88 — APPROVED** [review]

**Pull request merged**